### DISCUSSION TITLE
combining 2 plugins...!

### DISCUSSION THREAD

**Discussion by mary-9555** [jspsych/jsPsych#2156] [Q&A] [answered]
Hi guys.
I'm working on a task using multimedia such as images and audios.
After every trial there should be a Likert scale, for example in first page there's an image with "image_button_response" plugin used,
the following page there is a likert scale with "survey_likert" plugin, asking questions about the image.
So my question is, Should I define every single image variable then likert then image again then likert again and so on? or theres an easier way to code? Can I write a timeline_variables for my 72 trials! using two plugins?

I' m also wondering how can i present a trial including both image and audio at the same time in the same page ?

**Comment by jodeleeuw** [comment, marked as answer]
I'm moving this over to [discussions](https://github.com/jspsych/jsPsych/discussions).

**Comment by jodeleeuw** [comment]
Hi @mary-9555,

You can use timeline variables for this. You can have multiple trials on the timeline that the variables are applied to. Here's a quick example:

```js
var image_trial = {
	type: 'image-button-response',
	stimulus: jsPsych.timelineVariable('image'),
	choices: ['Continue']
}

var likert_trial = {
	type: 'survey-likert',
	questions: [
		{prompt: 'The image was large.', labels: ['Agree', 'Neutral', 'Disagree']},
		{prompt: 'The image was colorful.', labels: ['Agree', 'Neutral', 'Disagree']}
	]
}

var image_survey = {
	timeline: [image_trial, likert_trial],
	timeline_variables: [
		{image: '01.png'},
		{image: '02.png'},
		{image: '03.png'}
	]
}

jsPsych.init({
	timeline: [image_survey]
});
```

**Reply by mary-9555** [reply]
Thank you Josh!
I truly appreciate your help.
